pkg/api/validation/validation.go

```diff
@@ -273,6 +273,12 @@ func ValidateObjectMetaUpdate(new, old *api.ObjectMeta) errs.ValidationErrorList
 	} else {
 		new.CreationTimestamp = old.CreationTimestamp
 	}
+	// ignore changes to user name
+	if len(old.CreationUserName) == 0 {
+		old.CreationUserName = new.CreationUserName
```

Comment:
It was to allow an automated process to update it for legacy data.  So
someone upgrading from Kube 1.0 would be able to be assigned the value.  I
can disable the ability to set it post creation, which means we have no API
path to ever set that value and would have to write directly to etcd.
Allowing it to be set from empty to something post create has potential for
abuse.

On Sep 16, 2015, at 9:47 AM, David Eads <notifications@github.com> wrote:

In pkg/api/validation/validation.go
<https://github.com/kubernetes/kubernetes/pull/13925#discussion_r39630766>:

> @@ -273,6 +273,12 @@ func ValidateObjectMetaUpdate(new, old *api.ObjectMeta) errs.ValidationErrorList
>  	} else {
>  		new.CreationTimestamp = old.CreationTimestamp
>  	}
> +	// ignore changes to user name
> +	if len(old.CreationUserName) == 0 {
> +		old.CreationUserName = new.CreationUserName

If it was empty before, it seems weird to allow it to be updated later.

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/13925/files#r39630766>.